core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2137,13 +2137,14 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
           withMethodInputs(methodConfig, userInfo) { (wdl, methodInputs) =>
             withSubmissionEntityRecs(submissionRequest, workspaceContext, methodConfig.rootEntityType, dataAccess) { jobEntityRecs =>
               withWorkflowFailureMode(submissionRequest) { workflowFailureMode =>
+                val nonEmptyMethodInputs = methodInputs.filter(input => !input.workflowInput.optional || !input.expression.isEmpty)
```

Comment:
so these are the inputs that need to be evaluated, i.e. the ones that are non-optional or optional but filled in. maybe a comment to this effect?